app/assets/stylesheets/views/_homepage.scss

```diff
@@ -179,6 +173,16 @@
   @include govuk-media-query($from: "desktop") {
     @include columns($items: 16, $columns: 3, $selector: "li");
   }
+
+  .homepage-section__government-activity & {
```

Comment:
This is a very crafty solution but I wonder if it's very interpretable for devs. It does mean you're repeating yourself less by reusing those list reset styles but you're also effectively hiding the functionality inside a class that has a name tied to a different section. Maybe we could create generic list class to use between the sections but I'd be tempted to just repeat the CSS for this section and give it it's own class with a name that's explicit to that section. This is probably something for us to consider about if we can make this scss file a bit more efficient once we've deployed it.

I'm not gonna block this from going out if you don't agree as it's mostly a stylistic suggestion. I leave this to you. Happy to discuss.

Reply:
Thanks Owen, definitely understand the concerns here.

My thinking around this approach is that a parent selector requires less code to write but can help with readability as you can see in one block how the mixin can be modified based on the context - I think more comments will be needed though to explain that?

With 3 variants used across the home page it could be written something like this -

```
.homepage-generic-grid-name {
  @include columns($items: 16, $columns: 1, $selector: "li");
  list-style: none;
  margin: 0 govuk-spacing(-3);
  padding: 0;

  @include govuk-media-query($from: "tablet") {
    @include columns($items: 16, $columns: 2, $selector: "li");
  }

  @include govuk-media-query($from: "desktop") {
    @include columns($items: 16, $columns: 3, $selector: "li");
  }
  
  // Government activity
  .homepage-section__government-activity & {
    @include columns($items: 5, $columns: 1, $selector: "li");
    margin-bottom: govuk-spacing(4);

    @include govuk-media-query($from: "tablet") {
      @include columns($items: 5, $columns: 2, $selector: "li");
      margin-bottom: 0;
    }
  }
  
  // Most active
  .homepage-section__most-active & {
    @include govuk-media-query($from: desktop) {
      @include columns($items: 10, $columns: 2, $selector: "li", $flow: column);
    }
  }
}

I might leave things as they are for the moment... but definitely think we should revisit this when everything is pulled together in `new-homepage` and then decide on one approach 👍

Reply:
I can live with that. Are you ok to add this as a note to the [tech debt card](https://trello.com/c/RGlBCzuC/186-record-of-the-tech-debt-created-from-homepage-sprints-that-needs-clearing-up-post-deploy)?

Something else that occurs to me is that we could potentially filter all this grid stuff into the inevitable component so this ingests the grid values as well as the content to render this component. That would save all this grid chaff being needed here.

Reply:
Yep no worries (and I was thinking the same thing about the grid values too after i'd posted this). I'll update all on the tech debt card.